network/hybrid/network.go

```diff
@@ -242,37 +244,37 @@ func CreateNetworkInterface(cntx context.Context, netInterfaceName, nsgName, vne
 	if err != nil {
 		return nic, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, err))
 	}
-	err = future.WaitForCompletion(cntx, nicClient.Client)
+	err = future.WaitForCompletion(ctx, nicClient.Client)
 	if err != nil {
 		return nic, fmt.Errorf(fmt.Sprintf(errorPrefix, resourceName, fmt.Sprintf("cannot get nic create or update future response: %v", err)))
 	}
 	return future.Result(nicClient)
 }
 
 // GetNetworkSecurityGroup retrieves a netwrok resource group by its name
-func GetNetworkSecurityGroup(cntx context.Context, nsgName string) (network.SecurityGroup, error) {
-	environment, _ := azure.EnvironmentFromURL(helpers.ArmEndpoint())
+func GetNetworkSecurityGroup(ctx context.Context, nsgName string) (network.SecurityGroup, error) {
+	environment, _ := azure.EnvironmentFromURL(config.Environment().ResourceManagerEndpoint)
```

Comment:
Unless there's something I don't see, this should be simplified to:

``` Go
 environment := config.Environment()
```

(Same for all of the other instances in this file)